modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/portlet/action/ImportDataDefinitionMVCActionCommand.java

```diff
@@ -133,94 +142,146 @@ private UploadPortletRequest _getUploadPortletRequest(
 				liferayPortletRequest.getPortletName()));
 	}
 
+	private void _populateFieldNameBlacklist(
+		DataDefinition dataDefinition, Map<String, Boolean> fieldNameMap) {
+
+		for (DataDefinitionField dataDefinitionField :
+				dataDefinition.getDataDefinitionFields()) {
+
+			fieldNameMap.put(dataDefinitionField.getName(), true);
+		}
+	}
+
+	private void _populateFieldNameBlacklist(
+		DataDefinitionField dataDefinitionField,
+		Map<String, Boolean> fieldNameMap) {
+
+		for (DataDefinitionField nestedDataDefinitionField :
+				dataDefinitionField.getNestedDataDefinitionFields()) {
+
+			fieldNameMap.put(nestedDataDefinitionField.getName(), true);
+		}
+	}
+
 	private void _uniquifyDataDefinitionFields(DataDefinition dataDefinition) {
+		Map<String, Boolean> fieldNameMap = new HashMap<>();
```

Comment:
@rebsilva, once we're using this `HashMap` across several methods and, currently, it's being sent as a parameter from one method to other, wouldn't be better create it as an attribute of this class? What do you think?

Reply:
You are right, I'll do this!